test/native/demangleTest.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "arch.h"
+#include "testRunner.hpp"
+#include "demangle.h"
+#include <stdio.h>
+
+TEST_CASE(Demangle_test_needs_demangling) {
+    // Rust legacy-mangled symbol
+    CHECK_EQ(Demangle::needsDemangling("_ZN12panic_unwind3imp5panic17exception_cleanup17he4cf772173d90f46E"), true);
+    // Rust v0-mangled symbol
+    CHECK_EQ(Demangle::needsDemangling("_RNvCs6KtT2fMGqXk_8infiloop4main"), true);
+    // C++ symbol
+    CHECK_EQ(Demangle::needsDemangling("_ZNKSbIwSt11char_traitsIwESaIwEE4_Rep12_M_is_sharedEv"), true);
+    // C symbols
+    CHECK_EQ(Demangle::needsDemangling("malloc"), false);
+    CHECK_EQ(Demangle::needsDemangling("_malloc"), false);
+}
+
+// demangle calls _cxa_demangle, which should support demangling Itanium-mangled symbols across all Linux systems,
+// but might have a different mangling format on some other platforms.
+#ifdef __linux__
```

Comment:
What is Linux-specific here? Why can't it run on macOS?

Reply:
I'm not sure if the demangling scheme for MacOS is the same as the one for Linux.

Reply:
It certainly works on macOS, async-profiler uses it.